### ISSUE TITLE
TypeError when a dependency module depends upon Chalk - @rollup/plugin-commonjs > 23.0.4 

### ISSUE THREAD

**Issue by Jason3S** [rollup/plugins#1446]
- Rollup Plugin Name: @rollup/plugin-commonjs
- Rollup Plugin Version: >23.0.4
- Rollup Version: ^3.17.3
- Operating System (or Browser): node 
- Node Version: Node.js v16.14.2
- Link to reproduction _(⚠️ read below)_: https://stackblitz.com/edit/node-acyqrb?file=README.md

### Expected Behavior

The app not to crash on startup. See [repo](https://stackblitz.com/edit/node-acyqrb?file=README.md):

```sh
npm run test:ok
```

Result is as expected:

```
Hello Node.js v16.14.2!
{ config: {}, filepath: '/home/projects/node-acyqrb/test-cosmic.json' }
```

### Actual Behavior

```sh
npm run test:broken
```

Result:
```
TypeError: Cannot read properties of undefined (reading 'close')
    at eval (file://file:///home/projects/node-acyqrb/dist/index.mjs:2513:77)
    at requireChalk (file://file:///home/projects/node-acyqrb/dist/index.mjs:2670:3)
    at requireLib$1 (file://file:///home/projects/node-acyqrb/dist/index.mjs:2691:15)
    at requireLib (file://file:///home/projects/node-acyqrb/dist/index.mjs:2809:19)
    at requireParseJson (file://file:///home/projects/node-acyqrb/dist/index.mjs:2976:29)
    at Object.loadJson (file://file:///home/projects/node-acyqrb/dist/index.mjs:7141:17)
    at ExplorerSync.loadPackageProp (file://file:///home/projects/node-acyqrb/dist/index.mjs:7254:46)
    at ExplorerSync.loadFileContentSync (file://file:///home/projects/node-acyqrb/dist/index.mjs:7681:14)
    at ExplorerSync.createCosmiconfigResultSync (file://file:///home/projects/node-acyqrb/dist/index.mjs:7689:30)
    at ExplorerSync.loadSearchPlaceSync (file://file:///home/projects/node-acyqrb/dist/index.mjs:7666:17) 
```

### Additional Information

This is a bit of a nasty issue because it is related to a nested dependency:
```
devDependencies:
cosmiconfig 8.1.0
└─┬ parse-json 5.2.0
  └─┬ @babel/code-frame 7.18.6
    └─┬ @babel/highlight 7.18.6
      └─┬ chalk 2.4.2
        └── ansi-styles 3.2.1
```

It started when upgrading Cosminconfig from `8.0.0` to `8.1.0`.

See: https://github.com/cosmiconfig/cosmiconfig/issues/291
> I tracked this down to a change in `@rollup/plugin-commonjs`. Version `23.0.4` works, everything past it does not. It seems to be due to [this PR](https://github.com/rollup/plugins/pull/1363).
> 
> Please open an issue in [rollup's repository](https://github.com/rollup/plugins).
> 
> If they refuse to fix this for some reason, the package having to adapt would probably be `chalk` or `ansi-styles`, not `cosmiconfig`.
>
> _Originally posted by @d-fischer in https://github.com/cosmiconfig/cosmiconfig/issues/291#issuecomment-1445366612_

**Comment by coderaiser** [comment]
Using an alias with empty function helps, if `chalk` isn't used: 

```js
alias({
    entries: [
        {find: 'chalk', replacement: './lib/chalk.js'},
    ]
}),
```

**Comment by lukastaegert** [comment]
Fixed by #1455

**Comment by ThomasWhyne** [comment]
> Fixed by #1455

@lukastaegert  thanks for fixing, but why does this error reoccur in version `24.0.1`
